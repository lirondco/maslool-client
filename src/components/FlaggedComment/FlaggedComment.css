.FlaggedComment {
    margin-bottom: 2%;
    padding: 1%;
    border: #524c57 solid;
    border-radius: 10px;
    box-shadow: 5px 5px 2.5px grey;
  }

  .flagged_user {
      font-weight: bold;
      color: red;
  }

  .reported_by {
      font-style: italic;
  }
 
  .comment_content {
    padding: 2%;
    background-color: white;
    border: #524c57 solid 1px;
    border-radius: 10px;
    margin: 1%;
  }

  .action_buttons {
      display: flex;
      flex-direction: column;
      justify-content: left;
  }

  .FlaggedComment q {
      font-style: italic;
      background-color: white;
  }

  .FlaggedComment p {
      margin-left: 2%;
  }

  .FlaggedComment a p, .FlaggedComment button {
    border-radius: 8px;
    background-color: #524c57;
    color: #fff;
    font-weight: bolder;
    font-size: 12pt;
    padding: 4px 20px;
    border: 1px solid #CCC;
    box-shadow: 0 0 5px -1px rgba(0,0,0,0.2);
    width: max-content;
    margin-left: 2% !important;
    text-align: center;
  }

  .FlaggedComment a p:hover {
    background-color: brown;
  }
  