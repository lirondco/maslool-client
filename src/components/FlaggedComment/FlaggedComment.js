import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { Button } from '../Utils/Utils'
import './FlaggedComment.css'

export default class FlaggedComment extends Component {
    static defaultProps = {
        onBanClick: () => {}
    }

    handleBanClick = ev => {
        ev.preventDefault()
        window.confirm('Are you sure you want to do this?')
        this.props.onBanClick()
    }

    render() {
        const comment = this.props.comment 
        return (
            <div className='FlaggedComment'>
                <p className='flagged_user'>Comment Author: {comment.user.username} {comment.user.banned && '[banned]'}</p>
                <p className='reported_by'>Flagged by: {comment.flagged_by}</p>
                <p className='comment_content'>Comment: <q>{comment.content}</q></p>
                <div className='action_buttons'>
                    <Link to={`/trails/${comment.trail_id}`}><p className='comment_link'>Go to page</p></Link>
                    <Button type='button' onClick={this.handleBanClick}>{(comment.user.banned) ? 'Unban User' : 'Ban User'}</Button>
                </div>
            </div>
        )
    }
}