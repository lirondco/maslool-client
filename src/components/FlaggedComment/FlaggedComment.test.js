import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import FlaggedComment from './FlaggedComment'


describe(`Comments component`, () => {
    const props =
        {
            id: 9,
            content: "test comment test comment",
            last_modified: "2021-03-18T15:30:29.873Z",
            flagged: true,
            flaged_by: "test_flagger",
            user: {
                id: 3,
                username: "test_user",
                join_date: "2021-03-11T06:47:06.477Z",
                admin: true,
                banned: false
            }
        }



        it('renders a flagged comment', () => {
            const wrapper = shallow(<FlaggedComment comment={props} />)
            expect(toJson(wrapper)).toMatchSnapshot()
        })

        it('button works when clicked and window confirmation is rendered', () => {
            const wrapper = shallow(<FlaggedComment comment={props} />)
            window.confirm = jest.fn(() => true)
            wrapper.find('ForwardRef').simulate('click', { preventDefault() {}})
            expect(window.confirm).toBeCalled()
        })
})