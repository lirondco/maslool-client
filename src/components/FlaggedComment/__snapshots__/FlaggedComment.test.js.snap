// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments component renders a flagged comment 1`] = `
<div
  className="FlaggedComment"
>
  <p
    className="flagged_user"
  >
    Comment Author: 
    test_user
     
  </p>
  <p
    className="reported_by"
  >
    Flagged by: 
  </p>
  <p
    className="comment_content"
  >
    Comment: 
    <q>
      test comment test comment
    </q>
  </p>
  <div
    className="action_buttons"
  >
    <Link
      to="/trails/undefined"
    >
      <p
        className="comment_link"
      >
        Go to page
      </p>
    </Link>
    <ForwardRef
      onClick={[Function]}
      type="button"
    >
      Ban User
    </ForwardRef>
  </div>
</div>
`;
